graphs/eacs/src/test/kotlin/com/anaplan/engineering/azuki/graphs/eacs/BEH6.kt

```diff
@@ -0,0 +1,142 @@
+
+    @Eac("When a directed graph has one cycle, return true")
+    fun hasOneCycleDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("c", "e")
+            }
+        }
+        then {
+            hasCycle(graphA, true)
+        }
+    }
+
+    @Eac("When a directed graph has no cycle, return false")
+    fun hasNoCyclesDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+            }
+        }
+        then {
+            hasCycle(graphA, false)
+        }
+    }
+
+    @Eac("When a directed graph has a loop but one edge is in the inverse direction, return false")
+    fun hasReversedEdgeDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("e", "c")
+            }
+        }
+        then {
+            hasCycle(graphA, false)
+        }
+    }
+
+    @Eac("When a directed graph has multiple cycles with cycle edges shared, return true")
+    fun hasTwoCycleDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("c", "e")
+                edge("a", "f")
+                edge("f", "e")
+            }
+        }
+        then {
+            hasCycle(graphA, true)
+        }
+    }
+
+    @Eac("When a directed graph has multiple cycles with only one vertices shared, return true")
+    fun hasMultipleCycleDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("c", "e")
+                edge("a", "f")
+                edge("f", "g")
+                edge("h", "a")
+                edge("i", "h")
+                edge("g", "i")
+            }
+        }
+        then {
+            hasCycle(graphA, true)
+        }
+    }
+
+    @Eac("When a directed graph has multiple disconnected cycles, return true")
+    fun hasMultipleUnconnectedCycleDirected() {
+        given {
+            thereIsADirectedGraph(graphA) {
+                edge("a", "b")
+                edge("b", "c")
+                edge("d", "a")
+                edge("e", "d")
+                edge("c", "e")
+                edge("j", "f")
+                edge("f", "g")
+                edge("h", "j")
+                edge("i", "h")
+                edge("g", "i")
+            }
+        }
+        then {
+            hasCycle(graphA, true)
+        }
+    }
```

Comment:
🚫 These read like unit tests and I would argue that they are not acceptance criteria -- they are valuable analysis scenarios, though.